variables.tf

```diff
@@ -216,6 +216,16 @@ variable "cos_instance_name" {
   type        = string
   description = "The name to give the cloud object storage instance that will be provisioned by this module, required if 'var.provision_atracker_cos' is true."
   default     = null
+
+  validation {
+    condition     = var.cos_instance_name == null ? true : length(var.cos_instance_name) <= 180
+    error_message = "'var.cos_instance_name' must be 180 characters or less"
```

Comment:
I would drop the "var." and just have the message repeat the plain variable name.